probe-rs/src/architecture/arm/sequences/mod.rs

```diff
@@ -9,15 +9,27 @@ use std::{
     time::{Duration, Instant},
 };
 
+use probe_rs_target::CoreType;
+
 use crate::{architecture::arm::DapError, core::CoreRegister, DebugProbeError, Memory};
 
+use crate::architecture::arm::core::armv7a::Armv7DebugRegister;
+
 use super::{
     ap::AccessPortError,
     communication_interface::{DapProbe, Initialized},
     dp::{Abort, Ctrl, DpAccess, Select, DPIDR},
     ArmCommunicationInterface, DpAddress, Pins, PortType, Register,
 };
 
+/// An error occurred when executing an ARM debug sequence
+#[derive(thiserror::Error, Debug)]
+pub enum ArmDebugSequenceError {
+    /// Debug base address is required but not specified
+    #[error("Core access requries debug_base to be specified, but it is not")]
+    DebugBaseNotSpecified,
```

Comment:
Ah something like this is what I meant in the other comment on the session attach code where it is just an unwrap :)